src/highlight/highlightUtil.ts

```diff
@@ -58,3 +58,63 @@ export const centerHighlight = (shapes: Shape[]): Position => {
 export function isHighlight(annotation: Annotation): annotation is AnnotationHighlight {
     return annotation?.target?.type === Type.highlight;
 }
+
+export const getThreshold = (prev: number, curr: number, threshold: number): number =>
+    Math.max(Math.round(prev * threshold), Math.round(curr * threshold), 1);
+
+export const dedupeRects = (rects: Shape[], threshold = 0.1): Shape[] => {
+    const dedupedRects: Shape[] = [];
+
+    rects.forEach(curr => {
+        const prev = dedupedRects.pop();
+        // empty list, push current
+        if (!prev) {
+            dedupedRects.push(curr);
+            // The return in forEach callback is acting like a continue in for loop
+            return;
+        }
+
+        const { width: prevWidth, height: prevHeight, x: prevX, y: prevY } = prev;
+        const { width, height, x, y } = curr;
+
+        const xThreshold = getThreshold(prevWidth, width, threshold);
+        const yThreshold = getThreshold(prevHeight, height, threshold);
+
+        if (Math.abs(prevX - x) <= xThreshold && Math.abs(prevY - y) <= yThreshold) {
```

Comment:
Should width and height still be taken into account in determining duplication?

Reply:
If two rects have the same x and y, one must be the duplicate of the other, right? So I think only comparing x and y is enough for determining duplications.